rust/src/lib/ifaces/ipsec.rs

```diff
@@ -89,10 +89,79 @@ pub struct LibreswanConfig {
     pub ike: Option<String>,
     #[serde(skip_serializing_if = "Option::is_none")]
     pub esp: Option<String>,
+    #[serde(
+        skip_serializing_if = "Option::is_none",
+        default,
+        deserialize_with = "crate::deserializer::option_u64_or_string"
+    )]
+    pub dpddelay: Option<u64>,
+    #[serde(
+        skip_serializing_if = "Option::is_none",
+        default,
+        deserialize_with = "crate::deserializer::option_u64_or_string"
+    )]
+    pub dpdtimeout: Option<u64>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub dpdaction: Option<String>,
+    #[serde(
+        skip_serializing_if = "Option::is_none",
+        rename = "ipsec-interface",
+        default,
+        deserialize_with = "parse_ipsec_iface"
```

Comment:
Is not easier to make ipsec_interface a struct with a bool and a number ? instead of this multitype "yes", "no", number string ?

Reply:
@cathay4t do you have the NetworkManager ref around ?

Reply:
This is mainly for multi-compatibility. When used from network-role they can specify "1", 1, true, yes.. and so on. I know it is weird but it is a problem we need to solve. Similar for some bond options in our code

Reply:
https://gitlab.gnome.org/GNOME/NetworkManager-libreswan/-/merge_requests/25

Reply:
We don't have to follow exactly the same format, we can make it more "structural" like:
```yaml
use-ipsec-interfae: true|false
ipsec-interface-number: 33
```

Reply:
no. If you have, please state.

Reply:
I don't want to invent new stuff, just follow `ipsec.conf` manpage here. The section is named as `libreswan`, so I will refrain myself on abstracting stuff.

Reply:
This is patch to NetworkManager-libreswan for this : https://gitlab.gnome.org/GNOME/NetworkManager-libreswan/-/merge_requests/25/

I have scratch build in https://people.redhat.com/fge/ipsec_nmstate/

Reply:
Only support integer or string now. no bool support.